lib/Dialect/AIE/Transforms/AIEObjectFifoStatefulTransform.cpp

```diff
@@ -106,15 +106,20 @@ class DMAChannelAnalysis {
   int getDMAChannelIndex(TileOp tileOp, DMAChannelDir dir) {
     const auto &targetModel = getTargetModel(tileOp);
     int maxChannelNum = 0;
-    if (tileOp.isShimTile())
-      maxChannelNum = 2;
-    else {
-      if (dir == DMAChannelDir::MM2S)
-        maxChannelNum = targetModel.getNumSourceSwitchboxConnections(
-            tileOp.getCol(), tileOp.getRow(), WireBundle::DMA);
-      else
-        maxChannelNum = targetModel.getNumDestSwitchboxConnections(
-            tileOp.getCol(), tileOp.getRow(), WireBundle::DMA);
+    if (dir == DMAChannelDir::MM2S) {
+      maxChannelNum =
+          tileOp.isShimTile()
+              ? targetModel.getNumSourceShimMuxConnections(
+                    tileOp.getCol(), tileOp.getRow(), WireBundle::DMA)
+              : targetModel.getNumSourceSwitchboxConnections(
+                    tileOp.getCol(), tileOp.getRow(), WireBundle::DMA);
```

Comment:
Does `TileOp::getNumSourceConnections` / `TileOp::getNumDestConnections` work here?

Reply:
Ah I must've missed these! It's exactly what should be used. Let me update the PR. Thank you Jeff!

Reply:
Although I do find the comment regarding the reversed direction a bit confusing in these functions. For example, both of them use `getNumDestShimMuxConnections`..